File: tpFutbol5-domain-java7/src/ar/edu/futbol5/ordenamiento/OrdenamientoPorHandicap.java
package ar.edu.futbol5.ordenamiento;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

import ar.edu.futbol5.Jugador;
import ar.edu.futbol5.Partido;

public class OrdenamientoPorHandicap implements CriterioOrdenamiento {
	
	// BAD SMELL - Se repite el código de ordenar
	public List<Jugador> ordenar(Partido partido) {
		Collections.sort(partido.getInscriptos(), new Comparator<Jugador>() {
			@Override
			public int compare(Jugador jugador1, Jugador jugador2) {
				return calcularValor(jugador1).compareTo(calcularValor(jugador2));
			}
		});

		Collections.reverse(partido.getInscriptos());
		
		List<Jugador> jugadores=new ArrayList<Jugador>();
		jugadores.addAll(partido.getInscriptos());
		return jugadores;
	}
	
	public Double calcularValor(Jugador jugador) {
		return jugador.getCalificacion();
	}
	
}

File: tpFutbol5-domain-scala/src/ar/edu/futbol5/ordenamiento/OrdenamientoPorHandicap.scala
package ar.edu.futbol5.ordenamiento

import ar.edu.futbol5.Jugador
import ar.edu.futbol5.Partido

class OrdenamientoPorHandicap extends CriterioOrdenamiento {
  override def ordenar(partido: Partido): Seq[Jugador] = 
    partido.inscriptos.sortBy(calcularValor).clone.reverse

  override def calcularValor = _.calificacion 
}
